### DISCUSSION TITLE
v3.1.2

### DISCUSSION THREAD

**Discussion by assafelovic** [assafelovic/gpt-researcher#950] [Announcements]
Following the previous release, this release adds various bug fixes and performance improvements to the latest GPTR features including image support, documentation, new improved LLM and embedding configuration (shout out to @kesamet!). This release is recommended to all developers.

## What's Changed
* Sanitize filenames file upload/delete by @ethansilvas in https://github.com/assafelovic/gpt-researcher/pull/935
* "hack" to fix AzureOpenAI configuration problem by @danieldekay in https://github.com/assafelovic/gpt-researcher/pull/933
* Added a "Back To Top" Button by @Akhsuna07 in https://github.com/assafelovic/gpt-researcher/pull/937
* Specify embedding provider and model by @kesamet in https://github.com/assafelovic/gpt-researcher/pull/918
* added all changes by @assafelovic in https://github.com/assafelovic/gpt-researcher/pull/942
* Feature/strategic llm by @assafelovic in https://github.com/assafelovic/gpt-researcher/pull/941
* Docs upgrades by @ElishaKay in https://github.com/assafelovic/gpt-researcher/pull/940
* Fix/bing retriever consistency by @ibuder in https://github.com/assafelovic/gpt-researcher/pull/944
* fix/bing-retriever: add missing import statement by @rylincoln in https://github.com/assafelovic/gpt-researcher/pull/948

## New Contributors
* @ethansilvas made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/935
* @Akhsuna07 made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/937
* @ibuder made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/944
* @rylincoln made their first contribution in https://github.com/assafelovic/gpt-researcher/pull/948

**Full Changelog**: https://github.com/assafelovic/gpt-researcher/compare/v3.1.1...v3.1.2

<hr /><em>This discussion was created from the release <a href='https://github.com/assafelovic/gpt-researcher/releases/tag/v3.1.2'>v3.1.2</a>.</em>

**Comment by coreylansdell** [comment]
I have been unable to install this on my macbook pro. I've been trying for 2 days.

**Reply by danieldekay** [reply]
We are sorry to her that!

Can you please open an issue with your setup, how you are installing it and any error messages?

**Reply by coreylansdell** [reply]
Hey! I got it running late last night. I used Homebrew and followed the instructions found here https://www.youtube.com/watch?v=JVMl9rmP-eY&t=31s with the addition of including the API for Tavilly. Which the video does not describe. Thanks for getting back to me.
